### PR TITLE
Allow Mods to disable Popups of New Resistance Order Cards being available

### LINKED ISSUE #1126: Allow Mods to disable Popups of New Resistance Order Cards being available
We Already have an option to disable opening the monthly resistance order screen, but when you for example finish a covert action that increases faction influence, you still get a pop-up regarding new resistance cards being available. This is annyoying because I'd rather not get reminded of a garbage mechanic like that existing.
Together with #368 It should be possible to make resistance orders completely disappear from the game.

### PR DESCRIPTION
**Pull request by Grobobobo**
Fixes #1126

### CODE REVIEW

**Review by Iridar — CHANGES_REQUESTED** [review]
If this is meant as a global on/off switch for all new cards, whatever they may be, then an event is overkill. IMO this should be done as a config bool flag in CHHelpers.

**Comment by Grobobobo** [comment]
It's meant as an off switch in my particular case, Someone else may want more conditional changes. Of course I'm more than fine with making it a config variable myself.

**Comment by Grobobobo** [comment]
Looks like it works now

**Review by Iridar — APPROVED** [review]
LGTM

**Comment by Grobobobo** [comment]
Ah, wait, now I remember why I made this an event - We have a second wave that enables resistance orders, so the check was supposed to be based on whether that second wave is enabled. That's not possible with a config variable, so I think I should revert to the hook after all. Sorry about that.

**Comment by Iridar** [comment]
> Ah, wait, now I remember why I made this an event - We have a second wave that enables resistance orders, so the check was supposed to be based on whether that second wave is enabled. That's not possible with a config variable, so I think I should revert to the hook after all. Sorry about that.

If you build against Highlander, you can simply set `class'CHHelpers'.default.YourVarName = true;` in OPTC conditionally.

**Comment by Grobobobo** [comment]
Huh, I didn't know you could change config variables in runtime. Yeah, then current state is fine.

**Comment by Grobobobo** [comment]
Although won't you have to keep setting it every time you run the game? I guess it can be set every time on loaded save, and on campaign start.

**Comment by Iridar** [comment]
> Although won't you have to keep setting it every time you run the game? I guess it can be set every time on loaded save, and on campaign start.

Yeah that's what it would take.

**Comment by Iridar** [comment]
Per discussion with Xymanek on discord, it was decided that the event is the better approach. Sorry for the wasted work, it be like that sometimes.

**Comment by Grobobobo** [comment]
I've reverted the changes to the event.

**Comment by Iridar** [comment]
I think this is good to go, but I'll wait for Xym to confirm.

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_ResistanceFaction.uc [hunk 1]
```diff
@@ -1343,23 +1343,54 @@ function DisplayNewStrategyCardPopups()
 	local StateObjectReference CardRef;
 	local int idx;
 
-	if (NewPlayableCards.Length > 0)
+	//Issue #1126
+	if(ShouldDisplayNewCardPopUps())
 	{
-		// Queue popups for each new card received
-		foreach NewPlayableCards(CardRef, idx)
+		if (NewPlayableCards.Length > 0)
```
Comment (Xymanek) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_ResistanceFaction.uc:1349]:
```
if (NewPlayableCards.Length > 0 && /* Issue #1126 */ ShouldDisplayNewCardPopUps())
```

Please do not waste CPU time when there is no need for it

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_ResistanceFaction.uc [hunk 2]
```diff
@@ -1343,23 +1343,54 @@ function DisplayNewStrategyCardPopups()
...
+/// Allows overriding whether the new resistance order cards pop ups will be displayed.
+/// Default: Popups for new cards will be shown.
+///
+/// ```event
+/// EventID: UIPopups_DisplayNewCardPopups,
+/// EventData: [inout bool ShouldShow],
+/// EventSource: XComGameState_ResistanceFaction,
+/// NewGameState: none
+/// ```
+simulated protected function bool ShouldDisplayNewCardPopUps()
```
Comment (Xymanek) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_ResistanceFaction.uc:1378]:
```suggestion
simulated private function bool ShouldDisplayNewCardPopUps()
```

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_ResistanceFaction.uc [hunk 3]
```diff
@@ -1343,23 +1343,54 @@ function DisplayNewStrategyCardPopups()
...
+
+   Tuple = new class'XComLWTuple';
+   Tuple.Id = 'UIPopups_DisplayNewCardPopups';
+   Tuple.Data.Add(1);
+   Tuple.Data[0].kind = XComLWTVBool;
+   Tuple.Data[0].b = true;
+
+   `XEVENTMGR.TriggerEvent('UIPopups_DisplayNewCardPopups', Tuple, self, none);
+
+	return Tuple.Data[0].b;
```
Comment (Xymanek) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_ResistanceFaction.uc:1390]: Please fix indentation

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_ResistanceFaction.uc [hunk 4]
```diff
@@ -1343,23 +1343,54 @@ function DisplayNewStrategyCardPopups()
...
+{
+   local XComLWTuple Tuple;
+
+   Tuple = new class'XComLWTuple';
+   Tuple.Id = 'UIPopups_DisplayNewCardPopups';
+   Tuple.Data.Add(1);
+   Tuple.Data[0].kind = XComLWTVBool;
+   Tuple.Data[0].b = true;
+
+   `XEVENTMGR.TriggerEvent('UIPopups_DisplayNewCardPopups', Tuple, self, none);
```
Comment (Xymanek) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_ResistanceFaction.uc:1388]: While the current event name isn't problematic, something like `ShouldDisplayNewStrategyCardPopups` would make more sense IMO

**Review by Xymanek — CHANGES_REQUESTED** [review]

**Comment by Grobobobo** [comment]
I'm Unlikely to use this after all.

**Pull request closed without merging**